store/root.js

```diff
@@ -105,56 +103,31 @@ class Root extends Store {
   }
 
   @computed
-  get dataProviders() {
-    return this.user?.dataProviders
+  get dataProvider() {
+    return this.user.dataProvider
   }
 
   @computed
-  get baseUrl() {
-    if (!this.dataProvider) return null
-    return `/data-providers/${this.dataProvider.id}`
+  get dataProviders() {
+    return this.user?.dataProviders
   }
 
   @action async init(dataProviderId) {
     try {
-      this.user = new User()
+      this.user.onDataProviderChange(this.handleDataProviderChange.bind(this))
```

Comment:
I do not like fetching the user before calling `init()` method. That is why I did creating here.

If we have an init method for the initialisation, we should not run heavy operations like calling the API in the constructor.

Reply:
I am confused. API is not called in constructor. User and dataproviders are loaded when `.retrieve()` method is called. We don't use `new User({prefetch: true})`.

Reply:
Sorry, it's my fault.